ws-nextjs-app/pages/[service]/send/[id]/FormField/File/index.tsx

```diff
@@ -62,13 +63,15 @@ export default ({
       event.target.files,
     ) as File[];
 
-    const uploaded = [...filesInState];
+    let uploaded = [...filesInState];
 
     let liveRegionText = fileUploadLiveRegionText;
 
     chosenFiles.forEach(file => {
       uploaded.push({ file } as FileData);
-      liveRegionText = `${liveRegionText}${file.name}, `;
+      const renamedDuplicates = renameDuplicates(uploaded, liveRegionText);
```

Comment:
You could reduce the amount of looping you have to do by making this a recursive function, something like this:

```ts
    const duplicateFileNameCheck = (file, ogFileName, fileNameCount = 0) => {
      const newFile = new File(
        [file],
        fileNameCount ? `${ogFileName} (${fileNameCount})` : file.name,
        {
          type: file.type,
          lastModified: file.lastModified,
        },
      );

      if (
        uploaded.some(uploadedFile => {
          return newFile.name === uploadedFile.file.name;
        })
      ) {
        return duplicateFileNameCheck(newFile, ogFileName, fileNameCount + 1);
      }

      return newFile;
    };

    chosenFiles.forEach(file => {
      const checkedFile = duplicateFileNameCheck(file, file.name);
      uploaded.push({ file: checkedFile } as FileData);
      liveRegionText = `${liveRegionText}${checkedFile.name}, `;
    });
```

Reply:
nicee this is way neater ! implemented now,  good shout 😊